x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsServiceSettings.java

```diff
@@ -156,10 +164,23 @@ public OpenAiEmbeddingsServiceSettings(StreamInput in) throws IOException {
         } else {
             dimensionsSetByUser = false;
         }
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ML_MODEL_IN_SERVICE_SETTINGS)) {
+            modelId = in.readOptionalString();
```

Comment:
It is required but when streaming if reading from an older node there is no `modelId` to read. I set this to null as I could not think of a better default value but I've now change that to `modelId = "unset"`.

Either way with `null` or `"unset"` the inference request will fail, at least by using `"unset"` the string can be made required rather than optional. Because this action executes on the same node this code branch should never be hit anyway